sdk/search/azure-search/src/main/java/com/azure/search/models/SearchRequest.java

```diff
@@ -0,0 +1,593 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+
+/**
+ * Parameters for filtering, sorting, faceting, paging, and other search query
+ * behaviors.
+ */
+@Fluent
+public final class SearchRequest {
```

Comment:
Once we make `SearchDocumentsResult.getNextPageParameters` package-scope, we should be able to make this a package-scope implementation class too.